labman/db/process.py

```diff
@@ -1714,18 +1714,60 @@ def _parse_pico_csv(contents, sep='\t',
         pico_df: pandas DataFrame object
             DataFrame relating well location and DNA concentration
         """
-        raw_df = pd.read_csv(contents, sep=sep, skiprows=2, skipfooter=5,
-                             engine='python')
+
+        cleaned_contents = QuantificationProcess._rationalize_pico_csv_string(
+            contents)
+        contents_io = StringIO(cleaned_contents)
+
+        # when reading in concentrations, force them to come in as strings
+        # so can check for overflow entries using regex
+        raw_df = pd.read_csv(contents_io, sep=sep, skiprows=2, skipfooter=5,
+                             engine='python',
+                             converters={'[Concentration]': lambda x: str(x)})
 
         pico_df = raw_df[['Well', '[Concentration]']]
         pico_df = pico_df.rename(columns={'[Concentration]': conc_col_name})
 
+        # any concentrations containing strings of question marks
+        # (generated when you overflow the sensor; usually due to sample being
+        # too concentrated) should be replaced with the highest concentration
+        # found in this file, per wet lab practice. Start by
+        # getting mask of the concentration rows that hold only question marks;
+        # regex matches start of string followed by one or more literal
+        # question marks, followed by end of string
+        overflow_mask = pico_df[conc_col_name].str.contains(
+            r'^\?+$', regex=True)
+
         # coerce oddball concentrations to np.nan
         pico_df[conc_col_name] = pd.to_numeric(pico_df[conc_col_name],
                                                errors='coerce')
 
+        # find the highest concentration in the file and replace all overflow
+        # concentrations with that value
+        max_concentration = pico_df[conc_col_name].max()
+        pico_df.loc[overflow_mask, conc_col_name] = max_concentration
+
+        # if there are any NaN concentrations left, there's a problem with the
+        # parsing, so throw an error
+        if pico_df[conc_col_name].isnull().any():
+            raise ValueError("Some concentrations in pico green quantitation "
```

Comment:
Is this actually reported to user in the GUI? In other words, when you do this via the GUI and leave NaNs do you see this error?

----- As this is not part of this PR, if the answer is no; let's just open an issue.

Reply:
Thanks. Let's open an issue but make it the least priority possible.

Reply:
Will do ;)